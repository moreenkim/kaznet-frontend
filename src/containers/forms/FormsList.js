import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';
import Moment from 'react-moment';
import qs from 'qs';
import { Dropdown, DropdownToggle, DropdownItem, DropdownMenu } from 'reactstrap';

import '../LoadListAnimation.css';
import { withAlert } from 'react-alert';
import * as formActions from '../../store/forms/actions';
import * as formSelectors from '../../store/forms/reducer';
import * as globalSelectors from '../../store/global/reducer';
import * as globalActions from '../../store/global/actions';
import * as constants from '../../constants.js';
import * as errorHandlerSelectors from '../../store/errorHandler/reducer';
import ListView from '../../components/ListView';
import NoResults from '../../components/NoResults';
import ElementMap from '../ElementMap';
import MisconfiguredForm from '../../components/forms/MisconfiguredForm';

export class FormsList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false
    };
    this.handleChange = this.handleChange.bind(this);
  }

  async componentDidMount() {
    this.props.showListTitle();
    this.props.changePageTitle('Forms');
    this.props.changePageTitleButton('');
    this.props.changePageTarget('');
    let { search } = qs.parse(this.props.location.search.slice(1));
    const { page } = qs.parse(this.props.location.search.slice(1));
    const { has_task } = qs.parse(this.props.location.search.slice(1));
    let hasTask;

    if (search === undefined) {
      search = '';
    }
    this.props.searchVal(search);

    if (has_task === undefined) {
      hasTask = '';
    } else {
      hasTask = has_task;
    }

    this.props.getHasTask(hasTask);

    let pageNumber = Number(page);

    if (isNaN(pageNumber)) {
      pageNumber = 1;
    }
    await this.props.fetchForms(
      `${constants.API_ENDPOINT}/forms/?ordering=${
        constants.FORM_SORT_FIELD
      }&search=${search}&page=${pageNumber}&has_task=${hasTask}`
    );
    this.props.changePageNumber(pageNumber);
  }

  componentDidUpdate(prevProps) {
    let { search } = qs.parse(this.props.location.search.slice(1));
    if (search === undefined) {
      search = '';
    }

    const { has_task } = qs.parse(this.props.location.search.slice(1));
    let hasTask;
    if (has_task === undefined) {
      hasTask = '';
    } else {
      hasTask = has_task;
    }

    const { page } = qs.parse(this.props.location.search.slice(1));

    if (Number(page) !== this.props.currentPage && !isNaN(page)) {
      const pageNumber = Number(page);
      this.props.fetchForms(
        `${constants.API_ENDPOINT}/forms/?ordering=${
          constants.FORM_SORT_FIELD
        }&search=${search}&page=${pageNumber}&has_task=${hasTask}`
      );
      this.props.changePageNumber(pageNumber);
    }
    if (this.props.hasError !== prevProps.hasError) {
      if (this.props.hasError === true) {
        this.props.alert.show(this.props.errorMessage);
      }
    }
  }

  handleChange(e) {
    const hasTask = e.target.getAttribute('data-key');
    const param = `?has_task=${hasTask}`;
    if (hasTask === null) {
      this.setState({
        isOpen: !this.state.isOpen
      });
      return false;
    }
    const pageValue = this.props.pageParam;
    const searchString = this.props.searchParam;
    this.props.fetchForms(
      `${constants.API_ENDPOINT}/forms/${
        hasTask !== '' ? param : ''
      }&search=${searchString}&page=${pageValue}`
    );
    this.props.getHasTask(hasTask !== '' ? hasTask : '');
    this.setState({
      isOpen: !this.state.isOpen
    });
    return true;
  }

  render() {
    if (this.props.searchParam !== '' && this.props.formCount === null) {
      return this.renderLoading();
    }
    if (this.props.formCount === 0) {
      return <NoResults searchVal={this.props.searchParam} endpoint="forms" />;
    }
    if (this.props.rowsIdArray.length <= 0) return this.renderLoading();
    return (
      <div className="FormsList">
        <ListView
          renderHeaders={this.renderHeaders}
          rowsIdArray={this.props.rowsIdArray}
          rowsById={this.props.rowsById}
          renderRow={this.renderRow}
          endpoint="forms"
          pageLinks={this.props.pageLinks}
          totalPages={this.props.totalPages}
          currentPage={this.props.currentPage}
          firstPage={this.props.firstPage}
          lastPage={this.props.lastPage}
          searchVal={this.props.searchParam}
          sortField={constants.FORM_SORT_FIELD}
          sortOrder={constants.SORT_DESC}
          isFormPage
          isOpen={this.state.isOpen}
          handleChange={this.handleChange}
          hasTask={this.props.hasTask}
        />
      </div>
    );
  }

  renderLoading() {
    return (
      <center>
        <div className="lds-ripple">
          <div />
          <div />
        </div>
      </center>
    );
  }

  renderHeaders(firstPageLink) {
    const headerItems = [
      'Name',
      <Dropdown isOpen={this.isOpen} toggle={this.handleChange} key="dropdown">
        <DropdownToggle caret tag="span" className="nav-link">
          Task
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem key={0}>
            <Link to={`${firstPageLink}`} className="nav-link" key="1" data-key="2">
              All
            </Link>
          </DropdownItem>
          <DropdownItem key={1}>
            <Link to={`${firstPageLink}&has_task=true`} className="nav-link" key="1" data-key="1">
              Has Task
            </Link>
          </DropdownItem>
          <DropdownItem key={2}>
            <Link to={`${firstPageLink}&has_task=false`} className="nav-link" key="0" data-key="0">
              Does Not Have Task
            </Link>
          </DropdownItem>
        </DropdownMenu>
      </Dropdown>,
      'Last Modified'
    ];
    return <ElementMap items={headerItems} HTMLTag="th" />;
  }

  renderRow(row) {
    const rowItems = [
      <div key={row.id}>
        {row.attributes.title}
        {row.attributes.metadata.configuration_status !== constants.XFORM_CORRECTLY_CONFIGURED && (
          <MisconfiguredForm key={row.id} form={row} />
        )}
      </div>,
      row.attributes.task_name,
      <Moment key={row.id} format="DD-MM-YYYY HH:mm:ss">
        {row.attributes.modified}
      </Moment>
    ];
    return <ElementMap items={rowItems} HTMLTag="td" />;
  }
}

function mapStateToProps(state) {
  return {
    rowsById: formSelectors.getFormsById(state),
    rowsIdArray: formSelectors.getFormsIdArray(state),
    totalPages: formSelectors.getTotalPages(state),
    currentPage: formSelectors.getCurrentPage(state),
    pageLinks: formSelectors.getPageLinks(state),
    firstPage: formSelectors.getFirstPage(state),
    lastPage: formSelectors.getLastPage(state),
    searchParam: globalSelectors.getSearchValue(state),
    formCount: formSelectors.getTotalCount(state),
    hasError: errorHandlerSelectors.getHasError(state),
    errorMessage: errorHandlerSelectors.getErrorMessage(state),
    pageParam: globalSelectors.getPageNum(state),
    hasTask: formSelectors.getHasTask(state)
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      fetchForms: formActions.fetchForms,
      changePageNumber: formActions.changePageNumber,
      getHasTask: formActions.getHasTask,
      changePageTitle: globalActions.changePageTitle,
      changePageTitleButton: globalActions.changePageTitleButton,
      changePageTarget: globalActions.changePageTarget,
      showListTitle: globalActions.toggleDetailTitleOff,
      searchVal: globalActions.getSearchVal
    },
    dispatch
  );
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withAlert(FormsList));
