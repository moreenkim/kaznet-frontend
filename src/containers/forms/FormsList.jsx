import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';
import Moment from 'react-moment';
import qs from 'qs';
import { Dropdown, DropdownToggle, DropdownItem, DropdownMenu } from 'reactstrap';
import PropTypes from 'prop-types';

import { withAlert } from 'react-alert';
import * as formActions from '../../store/forms/actions';
import * as formSelectors from '../../store/forms/reducer';
import * as globalSelectors from '../../store/global/reducer';
import * as globalActions from '../../store/global/actions';
import {
  SORT_DESC,
  FORM_SORT_FIELD,
  XFORM_CORRECTLY_CONFIGURED,
  API_ENDPOINT
} from '../../constants';
import * as errorHandlerSelectors from '../../store/errorHandler/reducer';
import ListView from '../../components/ListView';
import NoResults from '../../components/NoResults';
import ElementMap from '../ElementMap';
import MisconfiguredForm from '../../components/forms/MisconfiguredForm';
import Loading from '../../components/global/Loading';

function renderRow(row) {
  const rowItems = [
    <div key={row.id}>
      {row.attributes.title}
      {row.attributes.metadata.configuration_status !== XFORM_CORRECTLY_CONFIGURED && (
        <MisconfiguredForm key={row.id} form={row} />
      )}
    </div>,
    String(row.attributes.metadata.downloadable),
    row.attributes.task_name,
    <Moment key={row.id} format="DD-MM-YYYY HH:mm:ss">
      {row.attributes.modified}
    </Moment>
  ];
  return <ElementMap items={rowItems} HTMLTag="td" />;
}

function getFetchURL(search, hasTask, pageNumber) {
  return `${API_ENDPOINT}/forms/?ordering=${FORM_SORT_FIELD}&search=${search}&has_task=${hasTask}&page=${pageNumber}`;
}

export class FormsList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false
    };
    this.handleChange = this.handleChange.bind(this);
  }

  async componentDidMount() {
    const {
      showListTitle,
      changePageTitle,
      changePageTitleButton,
      changePageTarget,
      searchVal,
      getHasTask,
      fetchForms,
      changePageNumber,
      location
    } = this.props;

    showListTitle();
    changePageTitle('Forms');
    changePageTitleButton('');
    changePageTarget('');

    const { search = '' } = qs.parse(location.search.slice(1));
    const { page = 1 } = qs.parse(location.search.slice(1));
    const { has_task: hasTask = '' } = qs.parse(location.search.slice(1));
    const pageNumber = Number(page);

    searchVal(search);
    getHasTask(hasTask);

    const url = getFetchURL(search, hasTask, pageNumber);

    await fetchForms(url);
    changePageNumber(pageNumber);
  }

  componentDidUpdate(prevProps) {
    const {
      location,
      changePageNumber,
      errorMessage = '',
      hasError,
      currentPage = 1,
      alert,
      fetchForms
    } = this.props;
    const { hasError: prevHasError } = prevProps;

    const { search = '' } = qs.parse(location.search.slice(1));
    const { page = 1 } = qs.parse(location.search.slice(1));
    const { has_task: hasTask = '' } = qs.parse(location.search.slice(1));
    const pageNumber = Number(page);

    if (pageNumber !== currentPage) {
      const url = getFetchURL(search, hasTask, pageNumber);
      fetchForms(url);
      changePageNumber(pageNumber);
    }
    if (hasError !== prevHasError && hasError === true && errorMessage !== '') {
      alert.show(errorMessage);
    }
  }

  handleChange(e) {
    const hasTaskValue = e.target.getAttribute('data-key');
    const { isOpen: oldIsOpen } = this.state;
    const { fetchForms, getHasTask, pageValue, searchString } = this.props;

    if (hasTaskValue === null) {
      this.setState({
        isOpen: !oldIsOpen
      });
      return false;
    }

    const url = getFetchURL(searchString, hasTaskValue, pageValue);

    fetchForms(url);

    getHasTask(hasTaskValue !== '' ? hasTaskValue : '');

    this.setState({
      isOpen: !oldIsOpen
    });

    return true;
  }

  renderHeaders(firstPageLink) {
    const headerItems = [
      'Name',
      'Active',
      <Dropdown isOpen={this.isOpen} toggle={this.handleChange} key="dropdown">
        <DropdownToggle caret tag="span" className="nav-link">
          Task
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem key={0}>
            <Link to={`${firstPageLink}`} className="nav-link" key="1" data-key="2">
              All
            </Link>
          </DropdownItem>
          <DropdownItem key={1}>
            <Link to={`${firstPageLink}&has_task=true`} className="nav-link" key="1" data-key="1">
              Has Task
            </Link>
          </DropdownItem>
          <DropdownItem key={2}>
            <Link to={`${firstPageLink}&has_task=false`} className="nav-link" key="0" data-key="0">
              Does Not Have Task
            </Link>
          </DropdownItem>
        </DropdownMenu>
      </Dropdown>,
      'Last Modified'
    ];
    return <ElementMap items={headerItems} HTMLTag="th" />;
  }

  render() {
    const {
      searchParam,
      formCount,
      hasTask,
      rowsIdArray,
      rowsById,
      lastPage,
      pageLinks,
      totalPages,
      currentPage,
      firstPage
    } = this.props;
    const { isOpen } = this.state;

    if (searchParam !== '' && formCount === null) {
      return <Loading />;
    }
    if (formCount === 0) {
      return <NoResults searchVal={searchParam} endpoint="forms" />;
    }
    if (rowsIdArray.length <= 0) return <Loading />;
    return (
      <div className="FormsList">
        <ListView
          renderHeaders={this.renderHeaders}
          rowsIdArray={rowsIdArray}
          rowsById={rowsById}
          renderRow={renderRow}
          endpoint="forms"
          pageLinks={pageLinks}
          totalPages={totalPages}
          currentPage={currentPage}
          firstPage={firstPage}
          lastPage={lastPage}
          searchVal={searchParam}
          sortField={FORM_SORT_FIELD}
          sortOrder={SORT_DESC}
          isFormPage
          isOpen={isOpen}
          handleChange={this.handleChange}
          hasTask={hasTask}
        />
      </div>
    );
  }
}

FormsList.propTypes = {
  rowsById: PropTypes.objectOf(PropTypes.object).isRequired,
  rowsIdArray: PropTypes.arrayOf(PropTypes.string).isRequired,
  formCount: PropTypes.number,
  hasError: PropTypes.bool,
  errorMessage: PropTypes.string,
  hasTask: PropTypes.string,
  alert: PropTypes.objectOf(PropTypes.any),
  // search stuff
  searchString: PropTypes.string,
  location: PropTypes.objectOf(PropTypes.any),
  searchParam: PropTypes.string.isRequired,
  // page stuff
  pageValue: PropTypes.number,
  totalPages: PropTypes.number,
  currentPage: PropTypes.number,
  firstPage: PropTypes.number,
  lastPage: PropTypes.number,
  pageLinks: PropTypes.objectOf(PropTypes.string).isRequired,
  // functions
  fetchForms: PropTypes.func.isRequired,
  changePageNumber: PropTypes.func.isRequired,
  getHasTask: PropTypes.func.isRequired,
  changePageTitle: PropTypes.func.isRequired,
  changePageTitleButton: PropTypes.func.isRequired,
  changePageTarget: PropTypes.func.isRequired,
  showListTitle: PropTypes.func.isRequired,
  searchVal: PropTypes.func.isRequired
};

FormsList.defaultProps = {
  pageValue: 1,
  searchString: '',
  alert: {},
  location: {},
  totalPages: 1,
  currentPage: 1,
  firstPage: 1,
  lastPage: 1,
  formCount: null,
  hasError: false,
  errorMessage: '',
  hasTask: ''
};

function mapStateToProps(state) {
  return {
    rowsById: formSelectors.getFormsById(state),
    rowsIdArray: formSelectors.getFormsIdArray(state),
    totalPages: formSelectors.getTotalPages(state),
    currentPage: formSelectors.getCurrentPage(state),
    pageLinks: formSelectors.getPageLinks(state),
    firstPage: formSelectors.getFirstPage(state),
    lastPage: formSelectors.getLastPage(state),
    searchParam: globalSelectors.getSearchValue(state),
    formCount: formSelectors.getTotalCount(state),
    hasError: errorHandlerSelectors.getHasError(state),
    errorMessage: errorHandlerSelectors.getErrorMessage(state),
    hasTask: formSelectors.getHasTask(state)
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      fetchForms: formActions.fetchForms,
      changePageNumber: formActions.changePageNumber,
      getHasTask: formActions.getHasTask,
      changePageTitle: globalActions.changePageTitle,
      changePageTitleButton: globalActions.changePageTitleButton,
      changePageTarget: globalActions.changePageTarget,
      showListTitle: globalActions.toggleDetailTitleOff,
      searchVal: globalActions.getSearchVal
    },
    dispatch
  );
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withAlert(FormsList));
