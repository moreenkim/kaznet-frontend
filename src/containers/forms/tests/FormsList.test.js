import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { Router } from "react-router";
import createBrowserHistory from "history/createBrowserHistory";
import moment from "moment-timezone";

import { FormsList } from "../FormsList";
import * as fixtures from "../../../store/forms/tests/fixtures";

const history = createBrowserHistory();

describe("containers/forms/FormsList", () => {
  it("renders without crashing", () => {
    shallow(
      <FormsList
        fetchForms={function() {}}
        getHasTask={function() {}}
        changePageTitle={function() {}}
        changePageTitleButton={function() {}}
        changePageTarget={function() {}}
        showListTitle={function() {}}
        changePageNumber={function() {}}
        location={history.location}
        searchVal={function() {}}
        rowsIdArray={fixtures.formsIdArray}
      />
    );
  });

  it("renders form list correctly", () => {
    moment.tz.setDefault("EAT");
    const wrapper = mount(
      <Router history={history}>
        <FormsList
          fetchForms={function() {}}
          getHasTask={function() {}}
          changePageTitle={function() {}}
          changePageTitleButton={function() {}}
          changePageTarget={function() {}}
          showListTitle={function() {}}
          rowsById={fixtures.formsById}
          rowsIdArray={fixtures.formsIdArray}
          endpoint={"forms"}
          pageLinks={fixtures.pageLinks}
          totalPages={fixtures.totalPages}
          currentPage={fixtures.currentPage}
          firstPage={fixtures.firstPage}
          changePageNumber={function() {}}
          lastPage={fixtures.lastPage}
          searchVal={function() {}}
          searchParam={fixtures.searchParam}
          location={history.location}
        />
      </Router>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
