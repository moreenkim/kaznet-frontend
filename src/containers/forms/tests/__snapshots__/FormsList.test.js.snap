// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/forms/FormsList renders form list correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <FormsList
    changePageNumber={[Function]}
    changePageTarget={[Function]}
    changePageTitle={[Function]}
    changePageTitleButton={[Function]}
    currentPage={1}
    endpoint="forms"
    fetchForms={[Function]}
    firstPage={1}
    lastPage={2}
    location={
      Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      }
    }
    pageLinks={
      Object {
        "first": "http://127.0.0.1:8000/api/v1/forms/?format=vnd.api%2Bjson&page=1",
        "last": "http://127.0.0.1:8000/api/v1/forms/?format=vnd.api%2Bjson&page=2",
        "next": "http://127.0.0.1:8000/api/v1/forms/?format=vnd.api%2Bjson&page=2",
        "prev": null,
      }
    }
    rowsById={
      Object {
        "1": Object {
          "attributes": Object {
            "created": "2018-06-21T12:15:23.170271+03:00",
            "deleted_at": null,
            "has_task": false,
            "id_string": "form_1",
            "last_updated": null,
            "modified": "2018-06-21T12:15:23.170313+03:00",
            "ona_pk": 1903,
            "project_id": 4328,
            "task_name": null,
            "title": "Form 1",
          },
          "id": "1",
          "type": "XForm",
        },
        "2": Object {
          "attributes": Object {
            "created": "2018-06-21T12:15:23.175198+03:00",
            "deleted_at": null,
            "has_task": false,
            "id_string": "form_2",
            "last_updated": null,
            "modified": "2018-06-21T12:15:23.175232+03:00",
            "ona_pk": 6373,
            "project_id": 5522,
            "task_name": null,
            "title": "Form 2",
          },
          "id": "2",
          "type": "XForm",
        },
        "3": Object {
          "attributes": Object {
            "created": "2018-06-21T12:15:23.175198+03:00",
            "deleted_at": null,
            "has_task": true,
            "id_string": "form_3",
            "last_updated": null,
            "modified": "2018-06-21T12:15:23.175232+03:00",
            "ona_pk": 6373,
            "project_id": 5522,
            "task_name": "Form 3 Task",
            "title": "Form 3",
          },
          "id": "3",
          "type": "XForm",
        },
      }
    }
    rowsIdArray={
      Array [
        "1",
        "2",
        "3",
      ]
    }
    searchVal={[Function]}
    showListTitle={[Function]}
    totalPages={2}
  >
    <div
      className="FormsList"
    >
      <ListView
        currentPage={1}
        endpoint="forms"
        firstPage={1}
        lastPage={2}
        pageLinks={
          Object {
            "first": "http://127.0.0.1:8000/api/v1/forms/?format=vnd.api%2Bjson&page=1",
            "last": "http://127.0.0.1:8000/api/v1/forms/?format=vnd.api%2Bjson&page=2",
            "next": "http://127.0.0.1:8000/api/v1/forms/?format=vnd.api%2Bjson&page=2",
            "prev": null,
          }
        }
        renderHeaders={[Function]}
        renderRow={[Function]}
        rowsById={
          Object {
            "1": Object {
              "attributes": Object {
                "created": "2018-06-21T12:15:23.170271+03:00",
                "deleted_at": null,
                "has_task": false,
                "id_string": "form_1",
                "last_updated": null,
                "modified": "2018-06-21T12:15:23.170313+03:00",
                "ona_pk": 1903,
                "project_id": 4328,
                "task_name": null,
                "title": "Form 1",
              },
              "id": "1",
              "type": "XForm",
            },
            "2": Object {
              "attributes": Object {
                "created": "2018-06-21T12:15:23.175198+03:00",
                "deleted_at": null,
                "has_task": false,
                "id_string": "form_2",
                "last_updated": null,
                "modified": "2018-06-21T12:15:23.175232+03:00",
                "ona_pk": 6373,
                "project_id": 5522,
                "task_name": null,
                "title": "Form 2",
              },
              "id": "2",
              "type": "XForm",
            },
            "3": Object {
              "attributes": Object {
                "created": "2018-06-21T12:15:23.175198+03:00",
                "deleted_at": null,
                "has_task": true,
                "id_string": "form_3",
                "last_updated": null,
                "modified": "2018-06-21T12:15:23.175232+03:00",
                "ona_pk": 6373,
                "project_id": 5522,
                "task_name": "Form 3 Task",
                "title": "Form 3",
              },
              "id": "3",
              "type": "XForm",
            },
          }
        }
        rowsIdArray={
          Array [
            "1",
            "2",
            "3",
          ]
        }
        sortField="modified"
        sortOrder="desc"
        totalPages={2}
      >
        <div>
          <Table
            bordered={true}
            className="kaznet-table"
            responsiveTag="div"
            tag="table"
          >
            <table
              className="kaznet-table table table-bordered"
            >
              <thead>
                <tr>
                  <ElementMap
                    HTMLTag="th"
                    items={
                      Array [
                        "Name",
                        "Task",
                        "Last Modified",
                      ]
                    }
                  >
                    <th
                      key="0"
                    >
                      Name
                    </th>
                    <th
                      key="1"
                    >
                      Task
                    </th>
                    <th
                      key="2"
                    >
                      Last Modified
                    </th>
                  </ElementMap>
                </tr>
              </thead>
              <tbody>
                <tr
                  key="2"
                >
                  <ElementMap
                    HTMLTag="td"
                    items={
                      Array [
                        "Form 2",
                        null,
                        <t
                          ago={false}
                          calendar={false}
                          decimal={false}
                          element="time"
                          filter={[Function]}
                          format="DD-MM-YYYY HH:mm:ss"
                          fromNow={false}
                          interval={60000}
                          onChange={[Function]}
                          toNow={false}
                          unit={null}
                          unix={false}
                          utc={false}
                      >
                          2018-06-21T12:15:23.175232+03:00
                      </t>,
                      ]
                    }
                  >
                    <td
                      key="0"
                    >
                      Form 2
                    </td>
                    <td
                      key="1"
                    />
                    <td
                      key="2"
                    >
                      <t
                        ago={false}
                        calendar={false}
                        decimal={false}
                        element="time"
                        filter={[Function]}
                        format="DD-MM-YYYY HH:mm:ss"
                        fromNow={false}
                        interval={60000}
                        key="2"
                        onChange={[Function]}
                        toNow={false}
                        unit={null}
                        unix={false}
                        utc={false}
                      >
                        <time
                          dateTime={"2018-06-21T09:15:23.175Z"}
                        >
                          21-06-2018 12:15:23
                        </time>
                      </t>
                    </td>
                  </ElementMap>
                </tr>
                <tr
                  key="3"
                >
                  <ElementMap
                    HTMLTag="td"
                    items={
                      Array [
                        "Form 3",
                        "Form 3 Task",
                        <t
                          ago={false}
                          calendar={false}
                          decimal={false}
                          element="time"
                          filter={[Function]}
                          format="DD-MM-YYYY HH:mm:ss"
                          fromNow={false}
                          interval={60000}
                          onChange={[Function]}
                          toNow={false}
                          unit={null}
                          unix={false}
                          utc={false}
                      >
                          2018-06-21T12:15:23.175232+03:00
                      </t>,
                      ]
                    }
                  >
                    <td
                      key="0"
                    >
                      Form 3
                    </td>
                    <td
                      key="1"
                    >
                      Form 3 Task
                    </td>
                    <td
                      key="2"
                    >
                      <t
                        ago={false}
                        calendar={false}
                        decimal={false}
                        element="time"
                        filter={[Function]}
                        format="DD-MM-YYYY HH:mm:ss"
                        fromNow={false}
                        interval={60000}
                        key="3"
                        onChange={[Function]}
                        toNow={false}
                        unit={null}
                        unix={false}
                        utc={false}
                      >
                        <time
                          dateTime={"2018-06-21T09:15:23.175Z"}
                        >
                          21-06-2018 12:15:23
                        </time>
                      </t>
                    </td>
                  </ElementMap>
                </tr>
                <tr
                  key="1"
                >
                  <ElementMap
                    HTMLTag="td"
                    items={
                      Array [
                        "Form 1",
                        null,
                        <t
                          ago={false}
                          calendar={false}
                          decimal={false}
                          element="time"
                          filter={[Function]}
                          format="DD-MM-YYYY HH:mm:ss"
                          fromNow={false}
                          interval={60000}
                          onChange={[Function]}
                          toNow={false}
                          unit={null}
                          unix={false}
                          utc={false}
                      >
                          2018-06-21T12:15:23.170313+03:00
                      </t>,
                      ]
                    }
                  >
                    <td
                      key="0"
                    >
                      Form 1
                    </td>
                    <td
                      key="1"
                    />
                    <td
                      key="2"
                    >
                      <t
                        ago={false}
                        calendar={false}
                        decimal={false}
                        element="time"
                        filter={[Function]}
                        format="DD-MM-YYYY HH:mm:ss"
                        fromNow={false}
                        interval={60000}
                        key="1"
                        onChange={[Function]}
                        toNow={false}
                        unit={null}
                        unix={false}
                        utc={false}
                      >
                        <time
                          dateTime={"2018-06-21T09:15:23.170Z"}
                        >
                          21-06-2018 12:15:23
                        </time>
                      </t>
                    </td>
                  </ElementMap>
                </tr>
              </tbody>
            </table>
          </Table>
          <Pagination
            aria-label="Page navigation example"
            listTag="ul"
            tag="nav"
          >
            <nav
              aria-label="Page navigation example"
              className=""
            >
              <ul
                className="pagination"
              >
                <PaginationItem
                  disabled={false}
                  tag="li"
                >
                  <li
                    className="page-item"
                  >
                    <Link
                      aria-label="First"
                      className="page-link"
                      replace={false}
                      to="/forms/?search=&status=&page=1"
                    >
                      <a
                        aria-label="First"
                        className="page-link"
                        href="/forms/?search=&status=&page=1"
                        onClick={[Function]}
                      >
                        First
                      </a>
                    </Link>
                  </li>
                </PaginationItem>
                <PaginationItem
                  disabled={true}
                  tag="li"
                >
                  <li
                    className="page-item disabled"
                  >
                    <Link
                      aria-label="Previous"
                      className="page-link"
                      replace={false}
                      to="#"
                    >
                      <a
                        aria-label="Previous"
                        className="page-link"
                        href="/"
                        onClick={[Function]}
                      >
                        «
                      </a>
                    </Link>
                  </li>
                </PaginationItem>
                <PaginationItem
                  disabled={false}
                  tag="li"
                >
                  <li
                    className="page-item"
                  >
                    <Link
                      aria-label="Next"
                      className="page-link"
                      replace={false}
                      to="/forms/?search=&status=&page=2"
                    >
                      <a
                        aria-label="Next"
                        className="page-link"
                        href="/forms/?search=&status=&page=2"
                        onClick={[Function]}
                      >
                        »
                      </a>
                    </Link>
                  </li>
                </PaginationItem>
                <PaginationItem
                  disabled={false}
                  tag="li"
                >
                  <li
                    className="page-item"
                  >
                    <Link
                      aria-label="Last"
                      className="page-link"
                      replace={false}
                      to="/forms/?search=&status=&page=2"
                    >
                      <a
                        aria-label="Last"
                        className="page-link"
                        href="/forms/?search=&status=&page=2"
                        onClick={[Function]}
                      >
                        Last
                      </a>
                    </Link>
                  </li>
                </PaginationItem>
              </ul>
            </nav>
          </Pagination>
        </div>
      </ListView>
    </div>
  </FormsList>
</Router>
`;
